### ISSUE TITLE
Script to remove all target-stuff

### ISSUE THREAD

**Issue by Sophietje** [utwente-fmt/vercors#603]
It seems to happen quite often, when someone pulls vercors for updates, that (s)he needs to remove all target directories to get a correct compilation. Would it be possible to either make a script (or include in `sbt clean`?) so that we don't need to manually remove these files?

**Comment by Sophietje** [comment]
This should perhaps also include bin/.classpath

**Comment by bobismijnnaam** [comment]
From <https://www.scala-sbt.org/1.x/docs/Running.html#Common+commands>: 

> | Command | Description |
> |-- | --|
> |clean | Deletes all generated files (in the target directory). |

I'd say that clean should fix our problem but apparently not...?

**Comment by bobismijnnaam** [comment]
One clue: I noticed today that running clean only triggers recompilation of the main project ("vercors"), and not recompilation of the sub projects, nor the compiled class files of `build.sbt`. Might be relevant.

Or we can just roll our own clean target that deletes every "target" folder in the project. But we need to be sure that no-one ever creates a file called "target" if we do that...

**Comment by niomaster** [comment]
It's because our dependencies are not `aggregate`'ed

**Comment by bobismijnnaam** [comment]
So we have to roll our own because we fall outside the "standard" sbt mold?

**Comment by niomaster** [comment]
We can probably aggregate them instead, though I vaguely recall a reason why we do not do it. We should investigate it again.

**Comment by bobismijnnaam** [comment]
I applied the fix on PR https://github.com/utwente-fmt/vercors/pull/580, locally the building seems to work fine. From [the docs](https://www.scala-sbt.org/1.x/docs/Multi-Project.html#Aggregation) it seems to do exactly what we want, and it also actually deletes code.

The only line that bothers me is the following:

> Note: aggregation will run the aggregated tasks in parallel and with no defined ordering between them. 

I guess this might be problematic since we generate the parsers, so if that happens late the build will fail. Unfortunately I have a hard time finding more documentation about aggregation (I'm also surprised why `dependsOn` and `aggregate` are separate...?)

I'll do a few more test runs with different viper commits as dependencies, and the CI will have a stab at it as well.

**Comment by bobismijnnaam** [comment]
The easiest path forward is by adding the `.classpath` file to the `clean` task. This is straightforward, except for a tiny bug in sbt (sbt/sbt/issues/6395) (I think), so pending that we can add it if everyone else agrees it's a satisfactory workaround.

**Comment by niomaster** [comment]
The exact terminology escapes me, but I am pretty sure that the parser generators (defined as a sourceGenerator) will run before compilation, because it's not a task (but some other sbt word)

**Comment by bobismijnnaam** [comment]
Closing because 504b835d98242af01f2de8fe611cc86ef6747405 fixes the cleaning issue. Only the classfiles in the root project/ folder are not deleted yet, we should create a new issue if that's needed. I'm also creating a new issue for deletion of `.classpath` upon `sbt clean` (#631). This issue also closes #199.
